internal/api/delete_stage_v1alpha1.go

```diff
@@ -35,6 +37,32 @@ func (s *server) DeleteStage(
 		}
 		return nil, connect.NewError(connect.CodeInternal, err)
 	}
+
+	approvedFreightList := kargoapi.FreightList{}
+	err := s.listFreightFn(
+		ctx,
+		&approvedFreightList,
+		&client.ListOptions{
+			Namespace: req.Msg.GetProject(),
+			FieldSelector: fields.OneTermEqualSelector(
+				kubeclient.FreightApprovedForStagesIndexField,
+				stage.Name,
+			),
+		},
+	)
+
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, err)
+	}
+
+	for i := range approvedFreightList.Items {
+		freight := approvedFreightList.Items[i]
+		delete(freight.Status.ApprovedFor, stage.Name)
+		if err := s.client.Status().Update(ctx, &freight); err != nil {
+			return nil, connect.NewError(connect.CodeInternal, err)
+		}
```

Comment:
Same thing as [here](https://github.com/akuity/kargo/pull/1189/files#r1410074709) re: needing to use a patch, unfortunately.